File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let res = [];
        let hashMap = new Map();
        for (let num of nums) {
            hashMap.set(num, (hashMap.get(num) || 0) + 1);
        }
        hashMap = [...hashMap].sort((a, b) => b[1]-a[1])
        for(let i = 0; i < k; i++){
            res.push(hashMap[i][0]);
        }
        return res;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        hm = dict()
        for i, num in enumerate(nums):
            if num in hm:
                hm[num] += 1
            else:
                hm[num] = 1
            if hm[num] > 1:
                return True
        return False